File: packages/stack-node/templates/src/utils/responses.ts
import { Response } from "express";

/** Standard JSON error: {"error": {"message", "code"?}}. */
export function errorResponse(
  res: Response,
  message: string,
  status = 400,
  code?: string
): Response {
  const error: Record<string, unknown> = { message };
  if (code) error.code = code;
  return res.status(status).json({ error });
}

/** Standard JSON success response. */
export function successResponse(res: Response, data: unknown, status = 200): Response {
  return res.status(status).json(data);
}

File: packages/stack-flask/templates/app/utils/responses.py
"""Standard JSON response helpers. Use these for all route responses."""
from flask import jsonify


def error_response(message: str, status: int = 400, code: str | None = None):
    """Return a standardized JSON error: {"error": {"message", "code"?}}."""
    payload = {"error": {"message": message}}
    if code:
        payload["error"]["code"] = code
    return jsonify(payload), status


def success_response(data, status: int = 200):
    """Return a JSON success response."""
    return jsonify(data), status
